src/dbnode/persist/fs/migration/task_options.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package migration
+
+import (
+	"errors"
+
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/persist"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/storage"
+)
+
+var (
+	errNewMergerFnNotSet       = errors.New("newMergerFn not set")
+	errInfoFilesResultError    = errors.New("infoFileResult contains an error")
+	errNamespaceMetadataNotSet = errors.New("namespaceMetadata not set")
+	errPersistManagerNotSet    = errors.New("persistManager not set")
+	errStorageOptionsNotSet    = errors.New("storageOptions not set")
+	errFilesystemOptionsNotSet = errors.New("filesystemOptions not set")
+)
+
+type taskOptions struct {
+	newMergerFn       fs.NewMergerFn
+	infoFileResult    fs.ReadInfoFileResult
+	shard             uint32
+	namespaceMetadata namespace.Metadata
+	persistManager    persist.Manager
+	storageOptions    storage.Options
+	fsOpts            fs.Options
+}
+
+var _ TaskOptions = &taskOptions{}
```

Comment:
Can probably just remove this altogether since NewOptions will fail to compile

Reply:
Yeah, probably less important for an Options since it's obvious what it's implementing, but it can be useful for quick sight reads when it's slightly harder to tell what it's implementing :+1: